server/routes/curfewAddress/addResidentDetails.ts

```diff
@@ -34,45 +48,92 @@ export default class AddResidentDetailsRoutes {
 
       if (!forename) {
-        validationErrors.push({ field: 'forename', message: 'Enter a first name' })
+        validationErrors.push({ field: 'forename', message: 'Enter the main occupier’s first name' })
       }
 
       if (!surname) {
-        validationErrors.push({ field: 'surname', message: 'Enter a last name' })
+        validationErrors.push({ field: 'surname', message: 'Enter the main occupier’s last name' })
       }
 
       if (!relation) {
-        validationErrors.push({ field: 'relation', message: "Enter the resident's relation to the offender" })
+        validationErrors.push({
+          field: 'relation',
+          message: `Enter the main occupier’s relationship to ${prisonerName}`,
+        })
+      }
+
+      if (otherResident) {
+        otherResident.forEach((resident: OtherResident, index: number) => {
+          if (this.areAllFieldsEmpty(resident)) {
+            otherResident.splice(index, 1)
+          }
+          if (!!resident.forename || !!resident.surname || !!resident.relation || !!resident.age) {
+            const residentIndexMessage =
+              otherResident.length > 1 ? ` the ${getOrdinal(index + 1)} other resident’s` : ' the other resident’s'
+            if (!resident.forename) {
+              validationErrors.push({
+                field: `otherResident[${index}][forename]`,
+                message: `Enter${residentIndexMessage} first name`,
+              })
+            }
+
+            if (!resident.surname) {
+              validationErrors.push({
+                field: `otherResident[${index}][surname]`,
+                message: `Enter${residentIndexMessage} last name`,
+              })
+            }
+
+            if (!resident.relation) {
+              validationErrors.push({
+                field: `otherResident[${index}][relation]`,
+                message: `Enter${residentIndexMessage} relationship to ${prisonerName}`,
+              })
+            }
+
+            if (!resident.age) {
+              validationErrors.push({
+                field: `otherResident[${index}][age]`,
+                message: `Enter${residentIndexMessage} age`,
+              })
+            }
+          }
+        })
       }
 
       return validationErrors
     })
 
-    const { checkRequestId, prisonNumber } = req.params
-    const requestSummary = await this.addressService.getStandardAddressCheckRequest(
-      req?.middleware?.clientToken,
-      prisonNumber,
-      Number(checkRequestId),
-    )
-
-    await this.addressService.addResident(req?.middleware?.clientToken, prisonNumber, requestSummary.requestId, {
+    const mainResident = {
+      residentId,
       forename,
       surname,
       phoneNumber,
       relation,
-      dateOfBirth,
-      age,
       isMainResident: true,
-    })
+    } as _ResidentSummary
+
+    const otherResidents = otherResident
+      ? otherResident.map(this.transformToResidentSummary).filter((resident: _ResidentSummary | null) => resident)
```

Comment:
as suggested above - if you filter empty/missing in advance then you won't need to filter here